scripts/rare-x/ingest_rarex_dataset.py

```diff
@@ -0,0 +1,187 @@
+
+    response = requests.get(uri, headers=headers)
+    response_json = json.loads(response.text)
+    status_code = response.status_code
+
+    return status_code, response_json
+
+
+def get_job_status(job_id):
+    """Call retrieveJobStatus."""
+
+    # first check job status - retrieveJob
+    uri = f"https://data.terra.bio/api/repository/v1/jobs/{job_id}"
+    
+    headers = {"Authorization": "Bearer " + get_access_token(),
+               "accept": "application/json"}
+
+    response = requests.get(uri, headers=headers)
+    response_json = json.loads(response.text)
+    status_code = response.status_code
+
+    return status_code, response_json
+
+
+def monitor_ingest(request_body):
+    """Monitor status of a submitted ingestDataset request."""
+
+    # get job id from ingestDataset request body
+    job_id = request_body["id"]
+
+    # 202 = job is running as confirmed in ingest_dataset()
+    job_status_code, job_status_response = get_job_status(job_id)
+
+    while job_status_code == 202:           # while job is running
+        print(f"{job_id} --> running")
+        time.sleep(10) # wait 10 seconds before getting job status
+        job_status_code, job_status_response = get_job_status(job_id) # get updated status info
+
+    # job completes (‘failed’ or ‘succeeded’) with 200 status_code
+    # consider any combination other than 200 + succeeded as failure
+    if not (job_status_code == 200 and job_status_response["job_status"] == "succeeded"):
+        print(f"{job_id} --> failed")
+        # if failed, get the resulting error message
+        job_result_code, job_result_response = get_job_result(job_id)
+        raise ValueError(job_result_response)    
+
+    # job completes successfully
+    # success_code is 200 and "job_status" = is "succeeded"
+    print(f"{job_id} --> succeeded")
+    
+    # get final job response for successfully completed ingest
+    job_result_code, job_result_response = get_job_result(job_id)
+
+    return job_result_response
+
+
+def ingest_dataset(dataset_id, data):
+    """Load data into TDR with ingestDataset."""
+
+    uri = f"https://data.terra.bio/api/repository/v1/datasets/{dataset_id}/ingest"
+
+    headers = {"Authorization": "Bearer " + get_access_token(),
+               "accept": "application/json",
+               "Content-Type": "application/json"}
+
+    response = requests.post(uri, headers=headers, data=data)
+    status_code = response.status_code
+
+    if status_code != 202:  # if ingest start-up fails
+        raise ValueError(response.text)
+
+    # if ingest start-up succeeds
+    return json.loads(response.text)
+
+
+def create_ingest_dataset_request(ingest_filepath, target_table_name):
+    """Create the ingestDataset request body."""
+
+    load_dict = {"format": "json",
+                 "path": ingest_filepath,
+                 "table": target_table_name,
+                 "resolve_existing_files": "true",
+                 "updateStrategy": "replace"
```

Comment:
do you always want to use the replace strategy? could be nice to have this be an input to the function to allow for "merge" - otherwise I'd suggest changing the name of the function to explicitly indicate that you're calling the "replace" strategy

Reply:
yeah always replace for them

Reply:
cool, honestly I'd probably still make update_strategy an input but set the default so you don't have to change anything else, like: 

`def create_ingest_dataset_request(ingest_filepath, target_table_name, update_strategy="replace"):`

and then on this line `"updateStrategy": update_strategy`